linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/bash/shared.sh

```diff
@@ -2,7 +2,7 @@
 
 MOUNT_OPTION="nodev"
 # Create array of local non-root partitions
-readarray -t partitions_records < <(findmnt --mtab --raw --evaluate | grep "^/\w" | grep "\s/dev/\w")
+findmnt --mtab --raw --evaluate | grep "^/\w" | grep "\s/dev/\w" | readarray -t partitions_records
```

Comment:
This command is resulting in a empty variable in RHEL7, RHEL8 and RHEL9 and this is caused because using `|` makes the `readarray` command to execute in a sub-shell which finally somehow impacts on the content it reads. Probably this was the reason to use process substitution instead of pipe.

However, the #9812 showed a corner case where non-posix features are disabled and a posix compatible solution would be very good here. Fortunately, since the bash 4.2 there is the `lastpipe` option which solves the problem of the `readarray` in a sub-shell. So, please, includes the following line before the line with the `findmnt ... | readarray`.
```
shopt -s lastpipe
findmnt --mtab --raw --evaluate | grep "^/\w" | grep "\s/dev/\w" | readarray -t partitions_records
```
This should solve the problem if the remediation is executed by `oscap` but wouldn't work in a interactive shell. To make it work also in interactive shells, it is necessary to ensure the `monitor mode` is on for that session. This is done by `set +m`. I don't see a risk of leaving these options enabled in the session and is also possible they are already enabled by default in some environments. So, I wouldn't disabled them after the `findmnt` command.

I short, I would suggest you to update this by ensuring `lastpipe` and `monitor mode`. It would be something like this:
```
shopt -s lastpipe
set +m
findmnt --mtab --raw --evaluate | grep "^/\w" | grep "\s/dev/\w" | readarray -t partitions_records
```

Please, make sure to include a comment before these commands to clarify why it was done this way.

Reply:
Doh, of course, `readarray` is a builtin. How about?

```bash
readarray -t partitions_records <<< "$(findmnt --mtab --raw --evaluate | grep "^/\w" | grep "\s/dev/\w")"
```

Reply:
I can confirm this would also work on RHEL VMs, but would it work on Azure environment reported in the description? Unfortunately I am not able to test it there.

Reply:
@rossengeorgiev , can you confirm this would work in the Azure environment?

Reply:
@tylerjharden could you test in your Azure environment?

Reply:
@tylerjharden , could you confirm and update this PR, please?